apps/api/test/functional/deployments.spec.ts

```diff
@@ -460,4 +463,114 @@ describe("Deployments API", () => {
       });
     });
   });
+
+  describe("POST /v1/deployments/{dseq}/deposit", () => {
+    it("should deposit into a deployment successfully", async () => {
+      const { userApiKeySecret, wallets } = await mockUser();
+      const dseq = "1234";
+      setupDeploymentInfoMock(wallets, dseq);
+
+      const mockTxResult = {
+        code: 0,
+        hash: "test-hash",
+        rawLog: "success"
+      };
+
+      jest.spyOn(signerService, "executeDecodedTxByUserId").mockResolvedValueOnce(mockTxResult);
+
+      const response = await app.request(`/v1/deployments/${dseq}/deposit`, {
+        method: "POST",
+        body: JSON.stringify({
+          data: {
+            deposit: 5
+          }
+        }),
+        headers: new Headers({ "Content-Type": "application/json", "x-api-key": userApiKeySecret })
+      });
+
+      expect(response.status).toBe(200);
+    });
+
+    it("should return 404 if deployment does not exist", async () => {
+      const { userApiKeySecret } = await mockUser();
+      const dseq = "1234";
+
+      jest.spyOn(deploymentService, "findByOwnerAndDseq").mockRejectedValueOnce(new NotFound("Deployment not found"));
+
+      const response = await app.request(`/v1/deployments/${dseq}/deposit`, {
+        method: "POST",
+        body: JSON.stringify({
+          data: {
+            deposit: 5
+          }
+        }),
+        headers: new Headers({ "Content-Type": "application/json", "x-api-key": userApiKeySecret })
+      });
+
+      expect(response.status).toBe(404);
+      const result = await response.json();
+      expect(result).toEqual({
+        error: "NotFoundError",
+        message: "Deployment not found"
+      });
+    });
+
+    it("should return 500 if transaction fails", async () => {
```

Comment:
thought(non-blocking): usually on 500 error service must be restarted because 500 error is an unexpected crash in program behavior and we cannot trust program state anymore. 

So, either failure in signing transaction is expected error which should be handled with 503 for example

Reply:
What I want to say is that definition of 500 error is "unexpected error", if you test that you get 500 this means it's not a 500 error. Because the test implies that this thing is expected, it cannot be 500 error

Reply:
Well transactions can fail for many unknown reasons, so we should return 503 in the case of unknown tx fail?

Reply:
imo doesn't make sense to test :) it would make testing if it was some certain handled error

Reply:
Ok yea makes sense